File: installer/src/main/java/com/teamscale/profiler/installer/PermissionError.java
package com.teamscale.profiler.installer;

/**
 * Fatal error that aborts the installation process due to insufficient permissions for the installer process. E.g.
 * can't create or write files.
 * <p>
 * Throwing this error causes a message to be printed that the user should try running the installer with
 * root/Administrator permissions.
 */
public class PermissionError extends FatalInstallerError {

	public PermissionError(String message) {
		super(message);
	}

	public PermissionError(String message, Throwable throwable) {
		super(message, throwable);
	}

}

File: installer/src/main/kotlin/com/teamscale/profiler/installer/PermissionError.kt
package com.teamscale.profiler.installer

/**
 * Fatal error that aborts the installation process due to insufficient permissions for the installer process. E.g.
 * can't create or write files.
 *
 *
 * Throwing this error causes a message to be printed that the user should try running the installer with
 * root/Administrator permissions.
 */
class PermissionError : FatalInstallerError {
	constructor(message: String) : super(message)

	constructor(message: String, throwable: Throwable) : super(message, throwable)
}
